airbyte-integrations/connectors/source-instagram/source_instagram/streams.py

```diff
@@ -251,7 +251,7 @@ def _state_has_legacy_format(self, state: Mapping[str, Any]) -> bool:
 
     def get_updated_state(self, current_stream_state: MutableMapping[str, Any], latest_record: Mapping[str, Any]):
         """Update stream state from latest record"""
-        record_value = latest_record[self.cursor_field]
+        record_value = latest_record.get(self.cursor_field, str(self._start_date))
         account_id = latest_record.get("business_account_id")
         state_value = current_stream_state.get(account_id, {}).get(self.cursor_field) or record_value
         max_cursor = max(pendulum.parse(state_value), pendulum.parse(record_value))
```

Comment:
As far as I can tell, if the `date` is missing in the previous record and current stream state, this will essentially cause a full refresh, since we'll output with the cursor being equal to the start date. 

That seems acceptable-ish as a stopgap but definitely not as a long term solution for an incremental stream. 

Is there any chance of this causing some sort of endless loop where we keep re-syncing data since we keep outputting state messages that don't increase?

Reply:
I think this wouldn't be an infinite loop per say, but it just won't advance the state. So every future sync will sync that same data.